cql3/query_processor.cc

```diff
@@ -554,14 +554,14 @@ future<::shared_ptr<cql_transport::messages::result_message>> query_processor::e
 template<typename... Args>
 future<::shared_ptr<result_message>>
 query_processor::execute_maybe_with_guard(service::query_state& query_state, ::shared_ptr<cql_statement> statement, const query_options& options,
-    future<::shared_ptr<result_message>>(query_processor::*fn)(service::query_state&, ::shared_ptr<cql_statement>, const query_options&, std::optional<service::group0_guard>, Args...), Args... args) {
+    future<::shared_ptr<result_message>>(query_processor::*fn)(service::query_state&, ::shared_ptr<cql_statement>, const query_options&, std::optional<service::group0_guard>, Args...), const Args&... args) {
```

Comment:
This is changing all the arguments, not just the warning.

I didn't understand what copy you are eliminating.

Reply:
@avikivity , according to the issue: https://github.com/scylladb/scylladb/issues/20361
In query_processor.cc:571
`
 auto p = get_statement(query_string, query_state.get_client_state());
    auto statement = p->statement;
    const auto warnings = std::move(p->warnings);
`
The const auto warnings vector will contain a copy of the warnings vector, because p->warnings is a const std::vector<sstring>

The issue suggested to making `warnings` and the function parameter as const reference. So a copy doesn't get created during the line `const auto warnings = std::move(p->warnings)` , that is my understanding.

Reply:
Wouldn't it be enough to change it to `const auto& warnings = p>warnings;`? Assuming the lifetime is okay.

Forcing all the arguments to be const references is too much unintended consequences for that thing.

Reply:
Probably better would be to remove `const` from `prepared_statement::warnings`, remove the temporary `warnings` from the `execute_direct_without_checking_exception_message`  and move directly from the `p->warnings`.

Reply:
Thanks, yes this will take care of the lifetime.